perllib/FixMyStreet/App/Controller/Admin/AreaStats.pm

```diff
@@ -1,225 +1,18 @@
-        $by_category{$p->category}->{$meta_state} += $p->get_column('state_count');
-    }
-    $c->stash->{by_category} = \%by_category;
-
-    # problems this month by state
-    $c->stash->{$_} = 0 for values %$state_map;
-
-    $c->stash->{open} = $c->model('DB::Problem')->to_body(
-        $c->stash->{body}
-    )->search(
-        {
-            %area_param,
-            confirmed => { '>=' => $time },
-        }
-    )->count;
-
-    my $comments = $c->model('DB::Comment')->to_body(
-        $c->stash->{body}
-    )->search(
-        {
-            %$params,
-            'me.id' => { 'in' => \"(select min(id) from comment where me.problem_id=comment.problem_id and problem_state not in ('', 'confirmed') group by problem_state)" },
-        },
-        {
-            join     => 'problem',
-            group_by => [ 'problem_state' ],
-            select   => [ 'problem_state', { count => 'me.id' } ],
-            as       => [ qw/problem_state state_count/ ],
-        }
-    );
-
-    while (my $comment = $comments->next) {
-        my $meta_state = $state_map->{$comment->problem_state};
-        $c->stash->{$meta_state} += $comment->get_column('state_count');
-    }
-
-    $params = {
-        %area_param,
-        'me.confirmed' => { '>=', $time },
-    };
-
-    # problems this month by category
-    my $recent_problems = $c->model('DB::Problem')->to_body(
-        $c->stash->{body}
-    )->search(
-        $params,
-        {
-            group_by => [ 'category' ],
-            select   => [ 'category', { count => 'me.id' } ],
-            as       => [ qw/category category_count/ ],
-        }
-    );
-
-    while (my $p = $recent_problems->next) {
-        $recent_by_category{$p->category} += $p->get_column('category_count');
-    }
-    $c->stash->{recent_by_category} = \%recent_by_category;
-
-    # average time to state change in last month
-    $params = {
-        %area_param,
-        'problem.confirmed' => { '>=', $time },
-    };
-
-    $comments = $c->model('DB::Comment')->to_body(
-        $c->stash->{body}
-    )->search(
-        { %$params,
-            'me.id' => \"= (select min(id) from comment where me.problem_id=comment.problem_id)",
-            'me.problem_state' => { '!=' => 'confirmed' },
-        },
-        {
-            select   => [
-                { avg => { extract => "epoch from me.confirmed-problem.confirmed" } },
-            ],
-            as       => [ qw/time/ ],
-            join     => 'problem'
-        }
-    )->first;
-    my $raw_average = $comments->get_column('time');
-    if (defined $raw_average) {
-        $c->stash->{average} = int( $raw_average / 60 / 60 / 24 + 0.5 );
-    } else {
-        $c->stash->{average} = -1;
-    }
-}
-
-sub load_user_body : Private {
-    my ($self, $c, $body_id) = @_;
-
-    $c->stash->{body} = $c->model('DB::Body')->find($body_id)
-        or $c->detach( '/page_error_404_not_found' );
+    $c->res->redirect('/dashboard?body=' . $c->stash->{body_id});
 }
```

Comment:
Given all this does is redirect is there any point in keeping this, or the link in the admin navigation?

Reply:
No not really. Wondered if anyone might have bookmarked it, but meh, they can update.